drivers/lora/lora_basics_modem/lbm_common.c

```diff
@@ -0,0 +1,549 @@
+	if (status != RAL_STATUS_OK) {
+		modem_release(dev);
+		return -EIO;
+	}
+	return 0;
+}
+
+int lbm_lora_test_cw(const struct device *dev, uint32_t frequency, int8_t tx_power,
+		     uint16_t duration)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	ral_status_t status;
+	int ret = 0;
+
+	/* Ensure available, freed by op_done_work */
+	if (!modem_acquire(dev)) {
+		return -EBUSY;
+	}
+
+	/* Configure modem for CW */
+	lbm_driver_antenna_configure(dev, MODE_CW);
+	data->modem_mode = MODE_CW;
+
+	/* Invalidate stored config */
+	data->mod_params.sf = 0;
+
+	/* Configure continuous wave */
+	status = ral_set_pkt_type(&config->ralf.ral, RAL_PKT_TYPE_LORA);
+	if (status != RAL_STATUS_OK) {
+		return status;
+	}
+	status = ral_set_rf_freq(&config->ralf.ral, frequency);
+	if (status != RAL_STATUS_OK) {
+		return status;
+	}
+	status = ral_set_tx_cfg(&config->ralf.ral, tx_power, frequency);
+	if (status != RAL_STATUS_OK) {
+		return status;
+	}
+
+	/* Start the continuous wave transmission */
+	status = ral_set_tx_cw(&config->ralf.ral);
+	if (status != RAL_STATUS_OK) {
+		ret = -EIO;
+		goto release;
+	}
+
+	/* Schedule the end of the transmission */
+	k_work_reschedule(&data->op_done_work, K_MSEC(duration));
+	return 0;
+release:
+	modem_release(dev);
+	return ret;
+}
+
+static int op_done_sync_rx(const struct device *dev)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	ral_lora_rx_pkt_status_t pkt_status;
+	ral_status_t status;
+	int ret;
+
+	/* Retrieve packet information before putting modem into sleep mode */
+	status = ral_get_pkt_payload(&config->ralf.ral, data->rx_state.sync.msg_len,
+				     data->rx_state.sync.msg, &data->rx_state.sync.msg_len);
+	if (status == RAL_STATUS_OK) {
+		LOG_HEXDUMP_DBG(data->rx_state.sync.msg, data->rx_state.sync.msg_len, "RX");
+		ret = 0;
+	} else {
+		LOG_ERR("Failed to retrieve packet payload");
+		ret = -EIO;
+	}
+
+	status = ral_get_lora_rx_pkt_status(&config->ralf.ral, &pkt_status);
+	if (status == RAL_STATUS_OK) {
+		data->rx_state.sync.rssi_dbm = pkt_status.rssi_pkt_in_dbm;
+		data->rx_state.sync.snr_db = pkt_status.snr_pkt_in_db;
+	} else {
+		LOG_WRN("Failed to query packet signal stats");
+		data->rx_state.sync.rssi_dbm = INT16_MIN;
+		data->rx_state.sync.snr_db = INT8_MIN;
+	}
+
+	return ret;
+}
+
+static void op_done_async_rx(const struct device *dev)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	ral_lora_rx_pkt_status_t pkt_status;
+	uint8_t rx_buffer[CONFIG_LORA_BASICS_MODEM_ASYNC_RX_MAX_PAYLOAD];
+	ral_status_t status;
+	uint16_t size;
+
+	/* Retrieve the packet payload */
+	status = ral_get_pkt_payload(&config->ralf.ral, sizeof(rx_buffer), rx_buffer, &size);
+	if (status == RAL_STATUS_OK) {
+		LOG_HEXDUMP_DBG(rx_buffer, size, "RX");
+	} else {
```

Comment:
maybe swap the condition and put the `LOG_HEXDUMP_DBG` unindented, this branch with just the dbg looks weird